content/en/blog/2022-08-26-manage-kyverno-policies-as-ocirepositories/index.md

```diff
@@ -0,0 +1,207 @@
+```
+
+Now, we are ready to create an OCI image to store my Kyverno policies.
+
+> ⚠️  You can find all the code examples in [GitHub](https://github.com/developer-guy/flux-kyverno-policies).
+
+In order to do that, we will clone our repository that holds Kyverno policies and create an OCI image to store them
+
+```shell
+$ git clone https://github.com/developer-guy/my-kyverno-policies.git
+$ cd my-kyverno-policies
+```
+
+> ⚠️ We are expecting that some other team “DevSecOps” is responsible to maintain and publish the policies to GHCR
+
+Next, we will be applying cert-manager policies into the cluster as an example:
+
+```shell
+$ flux push artifact oci://ghcr.io/developer-guy/policies:v1.0.0 \
+  --path="." \
+  --source="$(git config --get remote.origin.url)" \
+  --revision="$(git branch --show-current)/$(git rev-parse HEAD)"
+► pushing artifact to ghcr.io/developer-guy/policies:v1.0.0
+✔ artifact successfully pushed to ghcr.io/developer-guy/policies@sha256:56e853e3c5c02139c840b7f5c89a02f63ede8dc498ed3925a52360032aa49e60
+```
+
+> ⚠️ **Note:** Don't forget to change the values with your own details!
+
+Last but not least, we need to create an OCIRepository source that points out to my OCI image:
+
+```yaml
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: OCIRepository
+metadata:
+  name: kyverno-policies
+  namespace: flux-system
+spec:
+  interval: 5m
+  url: oci://ghcr.io/developer-guy/policies
+  ref:
+    semver: "v1.x"
+  secretRef:
+    name: ghcr-auth
+---
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
+kind: Kustomization
+metadata:
+  name: kyverno-policies
+  namespace: flux-system
+spec:
+  sourceRef:
+    kind: OCIRepository
+    name: kyverno-policies
+  interval: 60m
+  retryInterval: 5m
+  path: ./
+  prune: true
+  wait: true
+  timeout: 2m
+  patches: # enforce all policies
+    - patch: |
+        - op: add
+          path: /spec/validationFailureAction
+          value: enforce
+      target:
+        kind: ClusterPolicy
+```
+
+I'd like to highlight some key points about the resources above. Here in `OCIRepository` resource, we are using [SemVer](https://fluxcd.io/docs/components/source/ocirepositories/#semver-example) to select the policies that we want to apply. `.spec.ref` is an optional field to specify the OCI reference to resolve and watch for changes. If not specified, the latest version of the repository will be used. You can reach out to the complete list of references supported in Flux, here is the [link](https://fluxcd.io/docs/components/source/ocirepositories/#reference) for you. Also, in the `Kustomization` resource, we are using `.spec.patches` to apply patches to the policies that we want to enforce. We are using `op: add` to add a new field to the policy. `path` is the path to the field that we want to add. `value` is the value of the field that we want to add. To get more detail about the `Patches`, please see the [link](https://fluxcd.io/docs/components/source/kustomization/#patches).
```

Comment:
"OCIRepository" is stylized here in code formatting but previously without. Convention is that it should be without.

Reply:
The sentence beginning "You can reach" with the link doesn't make sense. There are only three sub-fields supported in `spec.ref`. Suggest removal.

Reply:
Same comment as the first with "Kustomization" resource.

Reply:
The link to https://fluxcd.io/docs/components/source/kustomization/#patches is broken and I am not able to find where this should point.